src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java

```diff
@@ -126,60 +120,131 @@ public void publishDiagnostics(PublishDiagnosticsParams publishDiagnosticsParams
+            ApplicationManager.getApplication().invokeAndWait(task);
+
+            try {
+                exitCode = task.get();
+            } catch (InterruptedException | ExecutionException e) {
+                LOG.warn(e.getMessage());
+                exitCode = -1;
+            }
+
         } else {
-            icon = null;
-            LOG.warn("No message type for " + message);
-        }
 
-        List<String> titles = new ArrayList<>();
-        for (MessageActionItem item : actions) {
-            titles.add(item.getTitle());
-        }
-        FutureTask<Integer> task = new FutureTask<>(
-                () -> Messages.showDialog(message, title, (String[]) titles.toArray(), 0, icon));
-        ApplicationManager.getApplication().invokeAndWait(task);
-
-        int exitCode = 0;
-        try {
-            exitCode = task.get();
-        } catch (InterruptedException | ExecutionException e) {
-            LOG.warn(e.getMessage());
+            final Notification notification = STICKY_NOTIFICATION_GROUP.createNotification(title, null, message, getNotificationType(msgType));
+            final CompletableFuture<Integer> integerCompletableFuture = new CompletableFuture<>();
+            for (int i = 0, optionsSize = options.length; i < optionsSize; i++) {
+                int finalI = i;
+                notification.addAction(new NotificationAction(options[i]) {
+                    @Override
+                    public boolean isDumbAware() {
+                        return true;
+                    }
+
+                    @Override
+                    public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
+                        integerCompletableFuture.complete(finalI);
+                        notification.expire();
+                    }
+                });
+            }
+            notification.whenExpired(() -> {
+                if (!integerCompletableFuture.isDone()) {
+                    integerCompletableFuture.complete(-1);
+                }
+            });
+            notification.notify(context.getProject());
+
+            try {
+                exitCode = integerCompletableFuture.get();
+            } catch (InterruptedException | ExecutionException e) {
+                LOG.warn(e.getMessage());
+                exitCode = -1;
+            }
         }
+        return CompletableFuture.completedFuture(actions == null || exitCode < 0 ? null : actions.get(exitCode));
+    }
 
-        return CompletableFuture.completedFuture(new MessageActionItem(actions.get(exitCode).getTitle()));
+    @NotNull
+    protected NotificationType getNotificationType(@NotNull MessageType messageType) {
+        NotificationType type;
+        switch (messageType) {
+            case Warning:
+                type = NotificationType.WARNING;
+                break;
+            case Error:
+                type = NotificationType.ERROR;
```

Comment:
Shall we consider swapping `Warning` and `Error` cases, just be aligned with the java log level standards?